Feature/config deployment

ilveroluca opened the pull request:
This PR brings a significant list of ansible tasks to preconfigure the k8s cluster deployed by manage-cluster. In particular, the plays will:
  * Install tiller
  * Configure storage classes for cinder and local volumes (two for each, varying in reclaim policy: one Delete and one Retain)
  * Configure ephemeral storage on the worker nodes to be exposed as local volumes
  * Install the static local volume provisioner
  * Install the NFS server provisioner

As before, the playbook is included in the template generated by `manage-cluster` and is executed with the `config-cluster` command.

Smaller changes included in this PR are:
  * bump default cloud image to ubuntu-18.04
  * explicitly disable the deployment of helm and the static local volume provisioner by kubespray
  * add crs4's helm chart repository on `config-client`

ilveroluca committed:
Use Ansible `include_tasks` instead of the obsolete `include`

ilveroluca committed:
Modify playbook to format and mount ephemeral space

ilveroluca committed:
Define n. pvs as a variable; compress retain/scratch creation tasks

ilveroluca committed:
Partial work on preparing and configuring local storage

ilveroluca committed:
Fix wrong clusterrole name

ilveroluca committed:
Finish installation of helm and local vol provisioner

ilveroluca committed:
Bump version

ilveroluca committed:
Remove troubleshooting items from README

Those issues causing those problems are now automatically configured and
should no longer arise.

ilveroluca committed:
Explicitly turn off helm and LVP deployment in default template

ilveroluca committed:
Bump default ubuntu image version to 18.04

ilveroluca committed:
Add deployment of nfs-storage-provisioner to config-cluster playbook

ilveroluca committed:
comments

ilveroluca committed:
Add CRS4's helm chart repo to client's repo list

mdrio commented:
Docker image tdmproject/manage-cluster-tf:1.1rc needs to be pushed on Dockerhub

mdrio commented:
LGTM

mgaggero reviewed (dismissed):
LGTM

kikkomep commented on k8s-tools/manage-cluster:
```diff
@@ -158,12 +158,16 @@ function gen_template() {
+  debug_log "Setting cloud provider to openstack, turning on metrics, turning off deployment of helm and LV provisioner"
+  sed -i -e 's/^[#[:blank:]]*cloud_provider:.*$/cloud_provider: openstack/' "${CLUSTER_DIR}/tf/group_vars/all/all.yml"
+  sed -i \
+    -e 's/^[#[:blank:]]*metrics_server_enabled:.*$/metrics_server_enabled: true/' \
```
Please try the following variant which is compatible with `sed` on macOS:

```
sed -i -e \
       's/^#\{0,1\}[[:blank:]]*metrics_server_enabled:.*$/metrics_server_enabled: true/; 
        s/^#\{0,1\}[[:blank:]]*helm_enabled:.*$/helm_enabled: false/;
        s/^#\{0,1\}[[:blank:]]*local_volume_provisioner_enabled:.*$/local_volume_provisioner_enabled: false/' \
        "${CLUSTER_DIR}/tf/group_vars/k8s-cluster/addons.yml"
```

ilveroluca replied:
Thanks.  I applied your patch

ilveroluca committed:
Try to make sure that tiller is ready before using it to install charts

ilveroluca committed:
Clean-up deployments and PVCs before destroying deployment

This change is required to avoid leaking resources dynamically created
by kubernetes.  Our previous tear-down procedure simply ran terraform
delete, but terraform doesn't know about any resources that kubernetes
may have allocated.

ilveroluca committed:
Fix sed compatibility issue with Mac OSX

ilveroluca committed:
Bump version

simleo commented on k8s-tools/manage-cluster:
```diff
@@ -354,24 +359,59 @@ END
-function config_cluster() {
-  debug_log "config_cluster function"
+  if command -v helm >/dev/null; then
+    helm repo add crs4 https://crs4.github.io/helm-charts/
```
This breaks if helm is installed but not initialized:

```
Error: Couldn't load repositories file (/home/simleo/.helm/repository/repositories.yaml).
You might need to run `helm init` (or `helm init --client-only` if tiller is already installed)
```

kikkomep replied:
Right! I also suggest to perform `helm repo update` after the `crs4` repository has been added.

```
if command -v helm >/dev/null; then
   helm init --client-only
   helm repo add crs4 https://crs4.github.io/helm-charts/
   helm repo update
fi
```

ilveroluca replied:
Thanks for pointing out this problem.  I just fixed it with f379ae5

ilveroluca committed:
Fix bad handling of case where helm is installed but not initialized

kikkomep reviewed (approved):
LGTM
